recipes/full_finetune_distributed.py

```diff
@@ -367,8 +367,8 @@ def _setup_optimizer(
         optimizer = config.instantiate(cfg_optimizer, self._model.parameters())
 
         if opt_state_dict:
-            opt_state_dict = utils.transform_opt_state_dict(
-                opt_state_dict, self._model, optimizer
+            opt_state_dict = FSDP.optim_state_dict_to_load(
```

Comment:
?

Reply:
This is needless indirection imo. Aren't we always using FSDP in our distributed recipes?

Reply:
What about FSDP2, does this still apply?

Reply:
No there is other mapping that has to occur there. But I am explicitly not aiming to tackle that here since it is still a dev recipe

Reply:
As long as you don't believe we'll need to add this solution back soon if FSDP2 use expands a bunch.